src/renderer.ts

```diff
@@ -152,6 +158,504 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+          });
+        rect.exit().remove();
+      }
+    };
+  }
+
+
+  createHTML(col: MultiValueColumn, context: IDOMRenderContext): IHTMLCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const padding = context.option('rowPadding', 1);
+    const colorScale = col.getColor();
+
+    return {
+      template: `<div class="heatmapcell" style="top:${padding}px;">
+                                   </div>`,
+      update: (n: HTMLDivElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const div = g.selectAll('div').data(col.getValue(d.v, d.dataIndex));
+        div.enter().append('div');
+        div
+          .style({
+            'width': cellDimension + 'px',
+            'background-color': (d, i) => colorScale(d),
+            'height': context.rowHeight(i) + 'px',
+            'left': (d, i) => (i * cellDimension) + 'px'
+          });
+
+
+      }
+    };
+  }
+
+  createCanvas(col: MultiValueColumn, context: ICanvasRenderContext): ICanvasCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const padding = context.option('rowPadding', 1);
+    const colorScale = col.getColor();
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+
+      const data = col.getValue(d.v, d.dataIndex);
+      data.forEach((d: number, j: number) => {
+        const x = (j * cellDimension);
+        ctx.beginPath();
+        ctx.fillStyle = colorScale(d);
+        ctx.fillRect(x, padding, cellDimension, context.rowHeight(j));
+      });
+    };
+  }
+
+}
+
+
+class SparklineCellRenderer implements ICellRendererFactory {
+
+  createSVG(col: MultiValueColumn, context: IDOMRenderContext): ISVGCellRenderer {
+    const xScale: d3.scale.Linear<number, number> = col.getSparklineScale().xScale.range([0, col.getWidth()]);
+    const yScale: d3.scale.Linear<number, number> = col.getSparklineScale().yScale;
+    return {
+      template: `<path class="sparklinecell"></path>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        yScale.range([context.rowHeight(i), 0]);
+        const line = d3.svg.line<number>()
+          .x((d, j) => xScale(j))
+          .y((d, j) => yScale(d))
+          .interpolate('linear');
+        d3.select(n).attr('d', line(col.getValue(d.v, d.dataIndex)));
+
+      }
+    };
+  }
+
+  createCanvas(col: MultiValueColumn, context: ICanvasRenderContext): ICanvasCellRenderer {
+
+    const xScale: d3.scale.Linear<number, number> = col.getSparklineScale().xScale.range([0, col.getWidth()]);
+    const yScale: d3.scale.Linear<number, number> = col.getSparklineScale().yScale;
+
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+      const data = col.getValue(d.v, d.dataIndex);
+      let xpos, ypos;
+      yScale.range([context.rowHeight(i), 0]);
+      ctx.strokeStyle = 'black';
+      ctx.fillStyle = 'black';
+      data.forEach((d, i) => {
```

Comment:
I still don't understand, why one single path for all elments in data isn't enough